### PR TITLE
Using docker for build and tests

### PR DESCRIPTION
**Pull request by filiphaftek**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,14 @@
-FROM alpine
+FROM golang:1.9 AS golang-build
```
Comment (kszarek) [inline Dockerfile]: why not an alpine version?

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+RUN mkdir -p /go/src/github.com/AirHelp/rabbit-amazon-forwarder
+WORKDIR /go/src/github.com/AirHelp/rabbit-amazon-forwarder
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o rabbit-amazon-forwarder .
 
+FROM alpine
```
Comment (kszarek) [inline Dockerfile]: always latest?

File: Dockerfile [hunk 3]
```diff
@@ -1,9 +1,14 @@
-FROM alpine
+FROM golang:1.9 AS golang-build
+RUN mkdir -p /go/src/github.com/AirHelp/rabbit-amazon-forwarder
+WORKDIR /go/src/github.com/AirHelp/rabbit-amazon-forwarder
+COPY . .
```
Comment (kszarek) [inline Dockerfile]: no dockerignore in the project, you will copy git directory

File: Makefile [hunk 4]
```diff
@@ -1,29 +1,12 @@
...
+push: test build
 	docker push airhelp/rabbit-amazon-forwarder
 
-fmt:
-	@echo 'run Go autoformat'
-	@gofmt -w $(GOFMT_FILES)
-
-# vet runs the Go source code static analysis tool `vet` to find
-# any common errors.
-vet:
```
Comment (kszarek) [inline Makefile]: no  vet and fmt anymore?

File: Dockerfile (hunk 2 above)
Comment (filiphaftek) [reply Dockerfile]: We could freeze it.

File: Makefile (hunk 4 above)
Comment (filiphaftek) [reply Makefile]: Used in docker with mounted local code.

**Pull request merged**